src/plot.js

```diff
@@ -148,6 +173,24 @@ function defaultFilter(index, channels, values) {
   return index;
 }
 
+function addScaleChannels(scaleChannels, channels, options) {
+  for (const [, channel] of channels) {
+    const {scale} = channel;
+    if (scale !== undefined) {
+      const {percent, transform = percent ? x => x * 100 : undefined} = options[scale] || {};
+      if (transform != null) channel.value = Array.from(channel.value, transform);
+      if (scaleChannels.has(scale)) scaleChannels.get(scale).push(channel);
+      else scaleChannels.set(scale, [channel]);
+    }
+  }
```

Comment:
This is duplicating code from Plot.plot:

https://github.com/observablehq/plot/blob/352d08670b8dfda49f60f5595d079a7b9808399c/src/plot.js#L41-L49